### PR TITLE
Make simpla-admin configurable

### LINKED ISSUE #1: Add config
Right now `simpla-admin` is not configurable. Add the following global config (with defaults): 

```js
// Prompt for login if unauthed and entering edit mode
loginPrompt: true

// Bind edit mode to #edit
hashTracking: true

// Enable alt+e hotkey edit mode toggling
quickToggle: true

// Enable pointer events (clicks) in edit mode
pointerEvents: false
```

These options need to be set globally, since `simpla-admin` is self-importing. We should use static methods on a global `Simpla.SimplaAdmin` object to achieve this in a robust manner:

```js
// Set config
Simpa.SimplaAdmin.setConfig('opt', 'value');
Simpla.SimplaAdmin.setConfig({ opt: value, opt2: value });

// Get current config
Simpla.SimplaAdmin.getConfig('opt');
```

Setting up these global config tools will be easy, weaving config flags all through the utils/behaviors will be more of a PITA.

### PR DESCRIPTION
**Pull request by madeleineostoja**
Attaches the simpla-admin singleton to `window.SimplaAdmin` and makes all major functionality (hash tracking, login prompts, hotkeys) configurable

Fixes #1

### CODE REVIEW

**Comment by madeleineostoja** [comment]
@bedeoverend can you LGTM when you get a mo?

File: src/simpla-admin.html
Comment (bedeoverend) [inline src/simpla-admin.html:146]: Probably better to put this into `attached`. Polymer's `ready` is a bit weird, rely's on components in the `local` DOM to be attached, `attached` is more in line with standard platform.

File: src/simpla-admin.html
Comment (bedeoverend) [inline src/simpla-admin.html:124]: Not a big deal given this is a singleton, but best practice, this should be a function, so `() => ({})`

File: src/behaviors/loginPrompt.js
Comment (bedeoverend) [inline src/behaviors/loginPrompt.js]: This is just a bit confusing - I'm not sure what's going on here. Be worth renaming / restructuring a few things to make it clearer what's happening e.g. the `loginPrompt` param - only understood what it is by checking the params, but maybe something like `useLoginPrompt` or some such?

File: src/behaviors/loginPrompt.js
Comment (bedeoverend) [inline src/behaviors/loginPrompt.js]: I don't think `&& loginPrompt` is needed? `initialPrompt` is only called lower down in the `if (loginPrompt) { ... }` block, so `initialPrompt` doesn't need to check it again.

File: src/behaviors/loginPrompt.js
Comment (bedeoverend) [inline src/behaviors/loginPrompt.js]: This shouldn't be doing the checks for if it's a function yet - it shouldn't _know_ that `promptLogin` requires `simplaLogin.prompt` to be a function. Instead this check should be done inside the `promptLogin` function. 

Also using the `setTimeout` is pretty hairy - just assumes it'll be ok in 10ms. Ideally the `prompt` should just take a prop like `open` or something which can be set even if the element isn't upgraded yet. Or, do a `setInterval` which gets cleared once `prompt` exists on the function.

**Review by bedeoverend — CHANGES_REQUESTED** [review]
Just tidying stuff, haven't pulled down - but you're more used to this so will leave that to you.

File: src/simpla-admin.html
Comment (madeleineostoja) [reply src/simpla-admin.html:146]: Righto, just thought ready got called sooner and this is pretty fundamental (was in created, but Polymer didn't like me assigning properties during created)

File: src/behaviors/loginPrompt.js
Comment (madeleineostoja) [reply src/behaviors/loginPrompt.js]: Ah this is left over from when I was setting defaults in `value:`, will clean up

File: src/behaviors/loginPrompt.js
Comment (madeleineostoja) [reply src/behaviors/loginPrompt.js]: What? It's a recursive function, it's exactly the same as `setInterval`, just more concise.

I didn't think `simplaLogin.prompt` checking should be in `promptLogin`, because that's lumping that extra logic into something that doesn't need it most of the time - basically you're just saying combine these two functions, I thought it was good to split them up because they're doing different things - one is prompting for login with a true/false param, the other is checking if it's possible to prompt for login then prompting for login if so.

File: src/behaviors/loginPrompt.js
Comment (madeleineostoja) [reply src/behaviors/loginPrompt.js]: Well `loginPrompt`is the user configuration, so wouldn't want to rename that - for a start should be in the JSDoc block (not sure why ESlint didn't complain that it's not). Not sure how else to make it clearer?

**Comment by madeleineostoja** [comment]
replied

File: src/simpla-admin.html
Comment (madeleineostoja) [reply src/simpla-admin.html:146]: Calling this on `attached` breaks things, leaving it on `ready`

Comment (bedeoverend) [reply src/simpla-admin.html:146]: Attached I think is just more robust. `ready()` might happen before or after `attached`, the timing is just a bit more confusing.

File: src/behaviors/loginPrompt.js
Comment (seaneking) [inline src/behaviors/loginPrompt.js]: Actually looking at it again I see what you mean now, yep will add the prompt check into `promptLogin`

Comment (bedeoverend) [reply src/behaviors/loginPrompt.js]: Re setTimeout - sorry, misread that. cool 👍 

If `promptLogin`'s implementation changes, then the `initialPrompt` checks become useless, so they're pretty tightly bound. Also the only reason there's a check for if it can be done is because `promptLogin`'s implementation may fail at some point, but that's more a concern for `promptLogin` i.e. it should either tell you it can't do it's job, or it should just always be able to do it's job i.e. contain that repeating logic in itself.

Comment (madeleineostoja) [reply src/behaviors/loginPrompt.js]: Yeah fair enough, my main concern was with the recursive call and observers, but will try it out

File: src/simpla-admin.html
Comment (madeleineostoja) [reply src/simpla-admin.html:146]: welp, `ready` seems to work across browsers for me, `attached` fails everywhere. Can readdress later if it becomes an issue

File: src/behaviors/loginPrompt.js
Comment (bedeoverend) [reply src/behaviors/loginPrompt.js]: `loginPrompt` is the user configuration, but doesn't need to be the parameter name, no reason the parameter name can't be called something more understandable. If the function was called something like `loginPromptObserver` then having it called the property name would make sense. JSDoc would help, though better if clearer in the comments.

**Comment by bedeoverend** [comment]
Replied

File: src/behaviors/loginPrompt.js
Comment (madeleineostoja) [reply src/behaviors/loginPrompt.js]: fair enough, will rename the funciton. IMO it's even more confusing if you go changing the prop name in the argument, harder to follow what's going on from a code flow perspective. I know there's a property called `loginPrompt`, and looking at this function I can assume it uses that prop, without having to go back up and check the observers

Comment (bedeoverend) [reply src/behaviors/loginPrompt.js]: I think that's more dangerous though - it means that if you ever see that variable, you assume its the property, even though there's no guarantee, it's just assuming that people have followed a convention, which can't be strictly enforced anyway.

I know what you mean - but I think it's better to need to explicitly check the observers, or change the function name so it's purpose is clearly stated to observe a prop. In general, I'm a bit off the magic of complex observers - the idea of writing a function name with the properties required in it as a string, harks back to Angular magic

**Comment by madeleineostoja** [comment]
Okay updated all the things - did you want to give a final LGTM or happy for me to merge @bedeoverend ?

File: src/simpla-admin.html
Comment (bedeoverend) [reply src/simpla-admin.html:146]: Sounds good

**Review by bedeoverend — APPROVED** [review]

**Comment by bedeoverend** [comment]
Knock yourself out - merge.

**Pull request merged**
